### PR TITLE
[edits] additional types, formatting, json file

### PR DESCRIPTION
**Pull request by zeeshanlakhani**
Attempts to match grammar to
    https://github.com/oxidecomputer/p4/examples.

Includes:
  * `varbit` and `tuple` types
  * `const` defn, updates to package/parser/control alts
  * table_element updates
  * new "directions": `packet_in`, `packet_out`
  * added `binops`
  * changes to highlights.scm (for helix, others)
  * preproc updates
  * ci addition to parse examples

### CODE REVIEW

File: grammar.js [hunk 1]
```diff
@@ -1,311 +1,525 @@
...
+          optional("("),
+          choice($._type, $.type_identifier),
+          optional(")"),
+          $.identifier,
+          optional(","),
+        ),
+      ),
 
-    direction: $ => choice('in', 'out', 'inout'),
+    direction: (_) => choice("in", "out", "inout", "packet_in", "packet_out"),
```
Comment (rcgoodfellow) [inline grammar.js]: `packet_in` and `packet_out` are not directions, they're [extern types](https://p4.org/p4-spec/docs/p4-16-working-draft.html#sec-packet-data-extraction) so they should be categorized as types.

File: queries/highlights.scm [hunk 2]
```diff
@@ -20,18 +23,29 @@
...
 "const" @keyword
 "meters" @keyword
 "counters" @keyword
-(direction) @keyword
-"bool" @type
-(bit_type) @type
+"_" @keyword
+"true" @constant
+"false" @constant
+(direction) @constant
```
Comment (rcgoodfellow) [inline queries/highlights.scm]: Why change directions from keywords to constants?

Comment (zeeshanlakhani) [reply queries/highlights.scm]: I had seen this in another tree-sitter grammar around highlighting, but easy to fix back.

**Comment by zeeshanlakhani** [comment]
@rcgoodfellow am fixing one minor bug with defines, should have that soon in the midst of things.

File: queries/highlights.scm [hunk 3]
```diff
@@ -20,18 +23,30 @@
...
 "const" @keyword
 "meters" @keyword
 "counters" @keyword
-(direction) @keyword
-"bool" @type
-(bit_type) @type
+"_" @keyword
+"true" @constant
+"false" @constant
+(direction) @
```
Comment (rcgoodfellow) [inline queries/highlights.scm]: I don't think this is a valid highlighting directive. Helix does not accept it.

Comment (zeeshanlakhani) [reply queries/highlights.scm]: @rcgoodfellow that's meant to be keyword, not sure what happened. apologies.

**Comment by rcgoodfellow** [comment]
`packet_in` and `packet_out` are not being highlighted, looks like there are missing scm directives for those fixed types.

**Comment by zeeshanlakhani** [comment]
@rcgoodfellow made an update after doing some helixing (emacs doesn't use the .scm mode)

**Comment by zeeshanlakhani** [comment]
@rcgoodfellow updated preproc to @keyword.directive.

**Review by rcgoodfellow — APPROVED** [review]

**Pull request merged**